### PR TITLE
Validate accelerated networking

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/13475

This PR adds 2 new validating webhooks for:

- node pool CREATE
- node pool UPGRADE

The webhooks check that the acceleratedNetworking flag is never changed during an upgrade and that is set to true on new node pools only if the instance type supports it.

### CODE REVIEW

File: pkg/azuremachinepool/validate_azuremachinepool_create.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/v2/pkg/apis/infrastructure/v1alpha2"
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+	restclient "k8s.io/client-go/rest"
+	expcapzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	apiv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
```
Comment (fiunchinho) [inline pkg/azuremachinepool/validate_azuremachinepool_create.go:15]: We are using v1alpha3

Comment (whites11) [reply pkg/azuremachinepool/validate_azuremachinepool_create.go:15]: Maybe this is why it's not working fuck me

Comment (whites11) [reply pkg/azuremachinepool/validate_azuremachinepool_create.go]: bumped

File: pkg/azureupdate/validate_azureclusterconfig_test.go [hunk 2]
```diff
@@ -280,8 +280,8 @@ func getClusterConfigAdmissionRequest(oldVersion string, newVersion string) *v1b
 			Kind:    "AzureClusterUpgrade",
 		},
 		Resource: metav1.GroupVersionResource{
-			Version:  "provider.giantswarm.io/v1alpha1",
-			Resource: "azureconfigs",
+			Version:  "core.giantswarm.io/v1alpha1",
+			Resource: "azureclusterconfigs",
```
Comment (fiunchinho) [inline pkg/azureupdate/validate_azureclusterconfig_test.go:284]: HmmMmm... why is this needed?

Comment (whites11) [reply pkg/azureupdate/validate_azureclusterconfig_test.go:284]: it's unrelated, but it was an error in master I wanted to fix

**Review by fiunchinho — APPROVED** [review]
LGTM

File: pkg/azuremachinepool/validate_azuremachinepool_create.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+func (a *CreateValidator) Validate(ctx context.Context, request *v1beta1.AdmissionRequest) (bool, error) {
+	azureMPNewCR := &expcapzv1alpha3.AzureMachinePool{}
+	if _, _, err := validator.Deserializer.Decode(request.Object.Raw, nil, azureMPNewCR); err != nil {
+		return false, microerror.Maskf(parsingFailedError, "unable to parse azureMachinePool CR: %v", err)
+	}
+
+	// If the instance type is invalid, the following function returns an error.
+	capabilities, err := vmcapabilities.FromInstanceType(azureMPNewCR.Spec.Template.VMSize)
+	if err != nil {
+		return false, microerror.Maskf(invalidOperationError, "Instance type is invalid or unsupported")
```
Comment (nprokopic) [inline pkg/azuremachinepool/validate_azuremachinepool_create.go:71]: IMHO we should have this error in a separate `VMSize` property validation.

Comment (nprokopic) [reply pkg/azuremachinepool/validate_azuremachinepool_create.go:71]: Wait, sorry, this is a "general" azuremachinepool validation. All good :)

File: pkg/internal/vmcapabilities/vmcapabilities.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package vmcapabilities
+
+type Capabilities struct {
```
Comment (nprokopic) [inline pkg/internal/vmcapabilities/vmcapabilities.go:3]: Here's SKU cache from CAPZ which they use for the capability check https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloud/services/resourceskus/cache.go

**Comment by fiunchinho** [comment]
Superseeded by https://github.com/giantswarm/azure-admission-controller/pull/17
